### PR TITLE
fix: show empty-state message in Issue History popup when no notes exist

### LINKED ISSUE #3363: Issue History popup (Family History / Other Meds / Risk Factors) shows a blank page with no empty-state message
## Summary
Clicking **Family History**, **Other Meds**, or **Risk Factors** in the CPP section of the encounter opens a popup ("Note History") that renders only two headings and nothing else — no content, no message explaining why. It reads as a broken/blank page rather than "no history recorded yet."

## Steps to Reproduce
1. In a patient's encounter, click **Family History** (or Other Meds / Risk Factors) in the CPP panel.
2. This calls `showIssueHistory(demoNo, issueId)` → opens `CaseManagementEntry?method=issuehistory&demographicNo=<n>&issueIds=<id>`.
3. Page renders exactly:
   ```
   Family History as part of cpp History
   FAKE-Bethany FAKE-Abbott
   ```
   ...and nothing else. Same pattern for Other Meds / Risk Factors, with the topic name swapped in the heading.

## Example Images

Image

Image

Image

## Root Cause
`CaseManagementEntry2Action.issuehistory()` populates `request.setAttribute("history", history)` from `caseManagementMgr.getIssueHistory(issueIds, demono)`, filtered to unlocked notes. The result JSP, `showHistory.jsp`, does:
```jsp
<h3 style="text-align: center;">${title}</h3>
<h3 style="text-align: center;">${demoName}</h3>
<c:forEach var="note" items="${history}" varStatus="idx">
  ...renders each note...
</c:forEach>
```
When `history` is empty, the `<c:forEach>` simply renders nothing — there is no `<c:if test="${empty history}">` fallback message at all. I confirmed directly against the dev DB that `casemgmt_issue_notes` has zero rows for the "Family History" (issue id 70), "Other Meds" (65), and "Risk Factors" (71) system issues in this seed data — so the empty list here happens to be correct (no notes exist), but the JSP gives no indication of that, and will show the exact same bare/blank page for **any** issue with zero linked notes in any environment, not just these three.

## Suggested Fix Direction
Add an empty-state message to `showHistory.jsp` (e.g. "No history has been recorded for this issue.") when `history` is empty, so the page doesn't look broken when it's legitimately just empty.

## Environment
Reproduced on local devcontainer, `develop` @ `aea7dafa28`.

### PR DESCRIPTION
**Pull request by vjymisal0**
## Description

Clicking Family History / Other Meds / Risk Factors in the CPP panel opens the "Note History" popup (`CaseManagementEntry?method=issuehistory`), rendered by `showHistory.jsp`. When the returned `history` list is empty, the JSP's `<c:forEach>` simply renders nothing after the two headings, so the page looks broken instead of communicating "no history recorded."

I traced the data path end-to-end (`CaseManagementEntry2Action.issuehistory()` -> `CaseManagementManagerImpl.getIssueHistory()` -> `CaseManagementNoteDAOImpl.getIssueHistory()`) and confirmed the query and locked-note filtering are correct — an empty result for an issue with zero linked notes is legitimate, not a data bug. So the fix is scoped to the JSP's missing empty-state fallback, as the issue itself suggested.

## Changes

- `showHistory.jsp`: added a `<c:if test="${empty history}">` block that renders a localized message when there are no history entries, plus the `fmt:setBundle` declaration required to use `fmt:message` in this JSP.
- Added a new `encounter.issueHistory.empty` key to `oscarResources_en.properties` ("No history has been recorded for this issue.") and to the `es`/`fr`/`pl`/`pt_BR` bundles as English placeholders with `# TODO: translate`, per the i18n contribution guide.

## Related Issues

Fixes #3363

## How Was This Tested?

No devcontainer/Docker/Maven toolchain was available in this environment, so I could not build/deploy and click through the popup live, or run the JSP-compiling test suite. What I did verify:

- Read the actual `issuehistory()` action method and the `getIssueHistory` HQL query directly (not just the issue description) to confirm the empty list is a legitimate result, not a bug to fix upstream.
- Ran `scripts/check-i18n-properties.sh` locally — all locales report 100% key coverage, 0 missing, 0 orphaned after the new key was added.
- Manually reviewed the JSP change for correct JSTL syntax and consistency with the `fmt:setBundle` placement convention used in sibling JSPs (e.g. `ChartNotes.jsp`).

I could not find an existing automated test harness that exercises this specific JSP rendering path (`showHistory.jsp` / `issuehistory()`), so no new automated test was added — flagging this rather than fabricating a test run.

## Screenshots

N/A — could not run the app in this environment (no Docker/devcontainer available). Happy to add a screenshot if a reviewer can confirm on a running instance, or if guided to a way to run it headlessly here.

## Checklist

- [x] My commits are signed off for the [DCO](https://developercertificate.org/) (`git commit -s`)
- [x] My commits follow [Conventional Commits](https://www.conventionalcommits.org/) format
- [x] I have not included any patient data (PHI) in this PR
- [ ] I have added tests for new functionality, or this change doesn't need new tests — see note above; no applicable test harness found for this JSP path
- [x] I have read the [contributing guide](CONTRIBUTING.md)

## Summary by Sourcery

Add an empty-state message to the Issue History popup when no notes exist.

New Features:
- Display a localized "no history recorded" message in the Note History popup when the history list is empty.

Bug Fixes:
- Prevent the Issue History popup from appearing broken by explicitly handling the case where no history entries are returned.

Enhancements:
- Configure the showHistory.jsp view to use the shared oscarResources bundle for localized messages.

Chores:
- Add the encounter.issueHistory.empty localization key to all supported resource bundles, with English text and TODO translation markers where needed.

---
## Summary by cubic
Shows a localized empty-state message in the Issue History popup when an issue has no notes, replacing the previous blank body. Loads the `oscarResources` bundle and adds `encounter.issueHistory.empty` with translations for `en/es/fr/pl/pt_BR`.

- Review Notes
  - Change is limited to `showHistory.jsp`; controllers/DAOs are unchanged.
  - Verify the message appears for `CaseManagementEntry?method=issuehistory` when `history` is empty and renders correctly in all listed locales.

<sup>Written for commit a2e96d3a612ee993a1ca8061e2bebe88761c5963. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/carlos-emr/carlos/pull/3373?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

**Comment by vjymisal0** [comment]
Addressed CodeRabbit's review:

1. Added the JSP documentation header block to `showHistory.jsp` (purpose, callers, `@since`), matching the convention in sibling JSPs like `ChartNotesAjax.jsp`.
2. Replaced the `encounter.issueHistory.empty` English placeholders in the es/fr/pl/pt_BR bundles with real translations instead of TODO placeholders.

One part of the second finding I did not apply: the suggestion that every translated value must include the literal string "CARLOS"/"CARLOS EMR". I checked `oscarResources_en.properties` and that branding appears in only ~27 of ~11,865 keys (mostly titles/headers), not as a blanket rule for every user-facing string, and no repo doc (`CONTRIBUTING.md`, `docs/I18N-STANDARDS.md`) states such a requirement. I left the message as a plain, unbranded string consistent with neighboring keys like `encounter.history.title`. Happy to add branding if a maintainer confirms that's actually the house style.

File: src/main/webapp/WEB-INF/jsp/casemgmt/showHistory.jsp [hunk 1]
```diff
@@ -31,6 +31,7 @@
 <%@ taglib uri="jakarta.tags.functions" prefix="fn" %>
 <%@ taglib uri="carlos" prefix="carlos" %>
 <%@ include file="/WEB-INF/jsp/casemgmt/taglibs.jsp" %>
+<fmt:setBundle basename="oscarResources"/>
```
[the remark this answers is not in the document]
Comment (vjymisal0) [reply src/main/webapp/WEB-INF/jsp/casemgmt/showHistory.jsp:49]: Fixed in 2fc896d — added the required JSP documentation block (purpose, callers, request attributes, `@since`) before the taglib declarations.

File: src/main/webapp/WEB-INF/jsp/casemgmt/showHistory.jsp [hunk 2]
```diff
@@ -62,6 +63,9 @@
 <body>
     <h3 style="text-align: center;">${title}</h3>
     <h3 style="text-align: center;">${demoName}</h3>
+    <c:if test="${empty history}">
+        <p style="text-align: center;"><fmt:message key="encounter.issueHistory.empty"/></p>
+    </c:if>
```
[the remark this answers is not in the document]
Comment (vjymisal0) [reply src/main/webapp/WEB-INF/jsp/casemgmt/showHistory.jsp:83]: Real es/fr/pl/pt_BR translations added in 2fc896d (replacing the English placeholders), and `scripts/check-i18n-properties.sh` confirms 100% key parity across all locales. Held off on adding literal "CARLOS"/"CARLOS EMR" branding to this string specifically — that pattern appears in only ~27 of ~11,865 existing i18n keys and isn't documented as a requirement anywhere in the repo, so it looked like a false positive rather than an established convention. Happy to add it if a maintainer confirms it's actually required.

**Comment by CecilFoubert** [comment]
Thank you very much for your hard work @vjymisal0! I have dug into this issue and found it shares a common root with another, so I'm making a PR to fix them both, which does supersede your work. However, I have credited your work in my PR to keep your contribution recognized. We appreciate your assistance, thank you for your time.

**Comment by vjymisal0** [comment]
Thanks for taking this further and for the credit, @CecilFoubert — closing this one in favor of your PR since it covers the same root cause.
